src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Windows.cs

```diff
@@ -95,14 +95,17 @@ private static unsafe void CompletionPortCallback(uint errorCode, uint numBytes,
                     }
                     else
                     {
+                        bool addRefedNativeOverlapped = false;
                         try
                         {
+                            asyncResult._nativeOverlapped.DangerousAddRef(ref addRefedNativeOverlapped);
```

Comment:
Before it was there, it was just implicit as part of marshaling the _nativeOverlapped as a SafeHandle.  Now that the P/Invoke is taking the raw pointer instead of a SafeHandle, the runtime isn't doing this implicitly, so I added doing it explicitly.